plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/GoogleApiFactory.java

```diff
@@ -150,22 +152,42 @@ private void buildTransportForUrl(String url) throws URISyntaxException {
   }
 
   private Proxy createProxy(String url) throws URISyntaxException {
-    Preconditions.checkNotNull(proxyService, "proxyService is null");
     Preconditions.checkNotNull(googleApiUrls, "googleApiUrl is null");
+    Preconditions.checkArgument(!url.startsWith("http://"), "http is not supported schema");
+
+    if (proxyService == null) {
+      return Proxy.NO_PROXY;
+    }
 
     IProxyData[] proxyData = proxyService.select(new URI(url));
     for (final IProxyData iProxyData : proxyData) {
```

Comment:
I think @briandealwis was suggesting `iProxyData` --> `proxyDatum`. (Technically `Datum` is a singular form, but I guess nowadays people just use `data` for anything.)